src/main/java/io/confluent/kafkarest/AvroRestProducer.java

```diff
@@ -94,7 +94,7 @@ public void produce(ProduceTask task, String topic, Integer partition,
         // if there isn't a schema. Validation will have already checked that all the keys/values
         // were NullNodes.
         Object key = (keySchema != null ? AvroConverter.toAvro(record.getKey(), keySchema) : null);
-        Object value = (valueSchema != null
```

Comment:
Does `AvroConverter.toAvro` not currently support this? If the schema supports null as a type (i.e. it is always null or is a union with null as one of the types), shouldn't the `toAvro` conversion just work, converting the `null` `JsonNode` to a `null` `Object` value? A unit test would be helpful here.

In any case, it seems like that'd be a better location for this check since it would centralize the conversion and makes it unit-testable. Centralizing seems useful since we would have missed the previous line, which presumably should also get the same treatment.